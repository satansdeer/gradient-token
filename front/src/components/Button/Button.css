.Button {
  background-color: rgb(251, 163, 113);
  border-radius: 4px;
  border: 2px solid #0b0b0b;
  box-shadow: rgb(255, 132, 75) 0px 0px 0px 3px inset;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 15px 20px;
}

.Button.is-disabled {
  background-color: #ccc;
  border-color: #999;
  box-shadow: #aaa 0px 0px 0px 3px inset;
  color: #eee;
  pointer-events: none;
}

