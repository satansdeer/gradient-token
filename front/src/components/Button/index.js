import React from "react";
import PropTypes from "prop-types";
import "./Button.css";

const Button = ({ onClick, label, disabled }) => (
  <button onClick={onClick} className={`Button ${disabled && "is-disabled"}`}>
    {label}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool
};

export default Button;
