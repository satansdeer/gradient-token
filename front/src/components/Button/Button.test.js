import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Button from "./index";

describe("<Button />", () => {
  const label = "Test";
  const mockOnClick = jest.fn();
  const component = shallow(<Button label={label} onClick={mockOnClick} />);

  it("renders and matches the snapshot", () => {
    const component = renderer.create(<Button label="Test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains the supplied label", () => {
    expect(component.text()).toContain(label);
  });

  it("calls the supplied onClick callback when clicked", () => {
    component.simulate("click");
    expect(mockOnClick).toHaveBeenCalled();
  });
});
