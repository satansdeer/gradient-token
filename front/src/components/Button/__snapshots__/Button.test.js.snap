// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders and matches the snapshot 1`] = `
<button
  className="Button undefined"
  onClick={undefined}
>
  Test
</button>
`;
